src/core/model_manager/model_manager.cpp

```diff
@@ -82,25 +85,136 @@ nlohmann::json ModelManager::CallComplete(const std::string &prompt, const std::
     return content_str;
 }
 
-nlohmann::json ModelManager::CallEmbedding(const std::string &input, const std::string &model) {
-    // List of supported models
-    static const std::unordered_set<std::string> supported_models = {"text-embedding-3-small",
-                                                                     "text-embedding-3-large"};
+nlohmann::json ModelManager::AzureCallComplete(const std::string &prompt, const std::string &model,
+                                                const nlohmann::json &settings, const bool json_response) {
+    // Get API key from the environment variable
+    const char *api_key = AzureModelManager::get_azure_api_key ();
```

Comment:
I would add the runtime_error exception throwing to the functions. I can forsee any changes to the error requiring multiple places and this leads to repeated code across any azure calls.

Reply:
Okay sure will move the error to the correct API

Reply:
done